### PR TITLE
Update Wrapper sample to add support running as process 

### PR DESCRIPTION
**Pull request by emmayspark**
instead of adding a new folder, update the code base to maintain one project.

### CODE REVIEW

File: wrappingGsdk/fakegame/Program.cs [hunk 1]
```diff
@@ -23,6 +18,11 @@ static void Main(string[] args)
                     Console.WriteLine($"Argument: {arg}");
                 } 
             }
+
+            HTTP_PORT = int.Parse(args[2]);
```
Comment (dgkanatsios) [inline wrappingGsdk/fakegame/Program.cs:22]: does this work? HTTP_PORT is specified as const above. It shouldn't be a const now.

Comment (dgkanatsios) [inline wrappingGsdk/fakegame/Program.cs:22]: I'd make it a TryParse to show an error and exit if the argument is not a valid integer

File: wrappingGsdk/wrapper/Program.cs [hunk 2]
```diff
@@ -38,8 +44,40 @@ static void Main(string[] args)
...
+
+            // here we're starting the script that initiates the game process
+            if (activeConfig.TryGetValue(portName, out string listeningPortString))
+            {
+                GameserverSDK.LogMessage($"{portName}:{listeningPortString} was found in GSDK Confg Settings.");
+                _listeningPort = listeningPortString;
+            }
+            else
+            {
+                LogMessage($"Cannot find {portName} in GSDK Config Settings. Please make sure the MockAgent is running " +
```
Comment (dgkanatsios) [inline wrappingGsdk/wrapper/Program.cs:57]: We don't call it MockAgent anymore, it's LocalMultiplayerAgent

File: wrappingGsdk/wrapper/Program.cs [hunk 3]
```diff
@@ -38,8 +44,40 @@ static void Main(string[] args)
...
+                             $"and that the MultiplayerSettings.json file includes correct {portName} as a GamePort Name.");
```
Comment (dgkanatsios) [inline wrappingGsdk/wrapper/Program.cs:58]: Is it GamePort or RealPort?

File: wrappingGsdk/wrapper/Program.cs [hunk 4]
```diff
@@ -151,5 +189,54 @@ private static void LogMessage(string message)
...
+            {
+                Console.WriteLine(exp.ToString());
+            }
+            return commandOut;
+        }
+
+        public static Process GetProcessPortInUse()
+        {
+            // execute netstat command for the given port
+            string commandArgument = string.Format("/c netstat -an -o -p tcp|findstr \":{0}.*LISTENING\"", _listeningPort);
```
Comment (dgkanatsios) [inline wrappingGsdk/wrapper/Program.cs:213]: this command will fail on Linux. But we can leave it for a different PR, where we update this sample to work with Linux processes. In any case, please add a comment here that this works only for Windows

File: wrappingGsdk/wrapper/Program.cs [hunk 5]
```diff
@@ -38,8 +44,42 @@ static void Main(string[] args)
...
+            {
+                LogMessage($"Cannot find {portName} in GSDK Config Settings. Please make sure the MockAgent is running " +
+                             $"and that the MultiplayerSettings.json file includes correct {portName} as a GamePort Name.");
+                return;
+            }
+
+            Process processPortInUse = GetProcessPortInUse();
+
+            // port in use, something is wrong - terminating wrapper.
+            if (processPortInUse != null)
```
Comment (dgkanatsios) [inline wrappingGsdk/wrapper/Program.cs:65]: can we make this if {...} a different method? Like CheckIfPortIsUsed and if returns true, it exits the app? It is not necessary but I feel it will make the sample more clear

File: wrappingGsdk/wrapper/Program.cs [hunk 6]
```diff
@@ -38,8 +44,42 @@ static void Main(string[] args)
...
         public static void InitiateAndWaitForGameProcess(string gameserverExe, IEnumerable<string> args)
         {
-             // here we're starting the script that initiates the game process
-            gameProcess = StartProcess(gameserverExe, args);
+            activeConfig = GameserverSDK.getConfigSettings();
+
+            // here we're starting the script that initiates the game process
+            if (activeConfig.TryGetValue(portName, out string listeningPortString))
+            {
+                GameserverSDK.LogMessage($"{portName}:{listeningPortString} was found in GSDK Confg Settings.");
```
Comment (dgkanatsios) [inline wrappingGsdk/wrapper/Program.cs:52]: typo: Config

File: wrappingGsdk/wrapper/Program.cs [hunk 7]
```diff
@@ -38,8 +44,42 @@ static void Main(string[] args)
         // starts main game process and wait for it to complete
         public static void InitiateAndWaitForGameProcess(string gameserverExe, IEnumerable<string> args)
         {
-             // here we're starting the script that initiates the game process
-            gameProcess = StartProcess(gameserverExe, args);
+            activeConfig = GameserverSDK.getConfigSettings();
+
+            // here we're starting the script that initiates the game process
+            if (activeConfig.TryGetValue(portName, out string listeningPortString))
```
Comment (dgkanatsios) [inline wrappingGsdk/wrapper/Program.cs:50]: can you please add a comment here for the difference in port between process and container?

File: wrappingGsdk/fakegame/Program.cs (hunk 1 above)
Comment (emmayspark) [reply wrappingGsdk/fakegame/Program.cs:25]: Oh I didn't commit the latest version of file. sorry just updated it.

File: wrappingGsdk/wrapper/Program.cs (hunk 3 above)
Comment (emmayspark) [reply wrappingGsdk/wrapper/Program.cs:58]: We are using `game_port` as a port name for MPS samples, and using `RealPort` for our functional tests and GSDK samples (e.g WinRunner).
For the consistency of MPSSample, I will use game_port as port name if that makes sense.

File: wrappingGsdk/wrapper/Program.cs (hunk 7 above)
Comment (emmayspark) [reply wrappingGsdk/wrapper/Program.cs:55]: Thank you so much for the review, I applied feedbacks and updated it.

File: wrappingGsdk/wrapper/Program.cs (hunk 4 above)
Comment (emmayspark) [reply wrappingGsdk/wrapper/Program.cs]: Thanks - add the code that GetProcessPortInUse() is available only on Windows

File: wrappingGsdk/wrapper/Program.cs [hunk 8]
```diff
@@ -38,8 +45,31 @@ static void Main(string[] args)
         // starts main game process and wait for it to complete
         public static void InitiateAndWaitForGameProcess(string gameserverExe, IEnumerable<string> args)
         {
-             // here we're starting the script that initiates the game process
-            gameProcess = StartProcess(gameserverExe, args);
+            // Here we're starting the script that initiates the game process
+
+            activeConfig = GameserverSDK.getConfigSettings();
+
+            // When Wrapper is running in a container, Port Information (Port Name, Port Number, and Protocol) is already set as build configuration.
```
[the remark this answers is not in the document]
Comment (emmayspark) [reply wrappingGsdk/wrapper/Program.cs:52]: updated the ReadMe. Thanks for the review :D

File: wrappingGsdk/README.md [hunk 9]
```diff
@@ -29,7 +29,7 @@ As mentioned, if you're just evaluating the platform or don't have a game server
 
 ### Creating the zipped game assets archive
 
-#### If you are using build.ps1 script
+#### - If you are using build.ps1 script
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:32]: I don't think it's common to have dashes on subtitles

File: wrappingGsdk/README.md [hunk 10]
```diff
@@ -39,26 +39,63 @@ You can use the zipped file that was created from `build.ps1` script during your
...
 
 > A common mistake new users do is that they compress the `publish` directory. This may create issues if you use incorrect mapping, so we highly recommend you zip the **files** of the `publish` folder instead.
 
-#### Deploying on MPS
+### Deploying on MPS
 
-You can use our [public documentation](https://docs.microsoft.com/en-us/gaming/playfab/features/multiplayer/servers/deploying-playfab-multiplayer-server-builds) to upload your Build on the MPS service. During the creation of your Build you should upload the zipped game assets you created and use a StartGameCommand like this one:
+Once you create game asset archive, next step is to creat a Build to deploy the game asset.
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:50]: typo: create

File: wrappingGsdk/README.md [hunk 11]
```diff
@@ -39,26 +39,63 @@ You can use the zipped file that was created from `build.ps1` script during your
...
+ ( Replace fakegame.exe with the name of your game server executable
+- Port Name: game_port
+- Port Type: TCP
+- Port Number: 80
 
-You should use `port 80 TCP` in the Build configuration. Bear in mind that during the allocation (i.e. usage of RequestMultiplayerServer API) the port you will get to connect will be different than 80. This is becase MPS service will create a mapping between the Azure Load Balancer (that exposes your ports to the Public internet) to the game servers running on the Azure Virtual Machines.
 
-### Using a Linux Build 
+ 
+> Bear in mind that during the allocation (i.e. usage of RequestMultiplayerServer API) the port you will get to connect will be different than 80.
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:69]: nit: empty lines before

File: wrappingGsdk/README.md [hunk 12]
```diff
@@ -39,26 +39,63 @@ You can use the zipped file that was created from `build.ps1` script during your
...
+- Port Name: game_port
+- Port Type: TCP
+- Port Number: 80
 
-You should use `port 80 TCP` in the Build configuration. Bear in mind that during the allocation (i.e. usage of RequestMultiplayerServer API) the port you will get to connect will be different than 80. This is becase MPS service will create a mapping between the Azure Load Balancer (that exposes your ports to the Public internet) to the game servers running on the Azure Virtual Machines.
 
-### Using a Linux Build 
+ 
+> Bear in mind that during the allocation (i.e. usage of RequestMultiplayerServer API) the port you will get to connect will be different than 80. 
+This is becase MPS service will create a mapping between the Azure Load Balancer (that exposes your ports to the Public internet) to the game servers running on the Azure Virtual Machines.
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:70]: nit: because

File: wrappingGsdk/README.md [hunk 13]
```diff
@@ -39,26 +39,63 @@ You can use the zipped file that was created from `build.ps1` script during your
...
 
-You should use `port 80 TCP` in the Build configuration. Bear in mind that during the allocation (i.e. usage of RequestMultiplayerServer API) the port you will get to connect will be different than 80. This is becase MPS service will create a mapping between the Azure Load Balancer (that exposes your ports to the Public internet) to the game servers running on the Azure Virtual Machines.
 
-### Using a Linux Build 
+ 
+> Bear in mind that during the allocation (i.e. usage of RequestMultiplayerServer API) the port you will get to connect will be different than 80. 
+This is becase MPS service will create a mapping between the Azure Load Balancer (that exposes your ports to the Public internet) to the game servers running on the Azure Virtual Machines.   
+ 
+    
+#### Build Configuration Example 2: Deploy Wrapper sample as Process on Windows VM.
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:73]: nit: empty line after headers (####)

File: wrappingGsdk/README.md [hunk 14]
```diff
@@ -39,26 +39,63 @@ You can use the zipped file that was created from `build.ps1` script during your
...
+activeConfig = GameserverSDK.getConfigSettings();
+portName = "game_port" 
+
+if (activeConfig.TryGetValue(portName, out string listeningPortString))
+{
+    // ... more code here
+}
+else
+```
+If you replace port name in wrapper sample, make sure specify updated port name in Build Configuration.
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:94]: nit: "to specify"

Comment (dgkanatsios) [inline wrappingGsdk/README.md:94]: maybe mention that they need to change the portName variable value above

File: wrappingGsdk/README.md [hunk 15]
```diff
@@ -45,20 +45,52 @@ You should copy all your game server build files from wherever your `wrapper` ex
...
-```
-# replace fakegame.exe with the name of your game server executable
-C:\Assets\wrapper.exe -g C:\Assets\fakegame.exe arg1 arg2
+### Deploying on MPS
+
+Once you create game asset archive, next step is to create a Build to deploy the game asset.
+
+You can refer to our [public documentation](https://docs.microsoft.com/en-us/gaming/playfab/features/multiplayer/servers/deploying-playfab-multiplayer-server-builds) to upload your Build on the MPS service. 
+
+During the creation of your Build, you should upload the zipped game assets and specify build configuration such as StartGameCommand, Port, and Virtual Machine Type as bellow:
```
Comment (dgkanatsios) [inline wrappingGsdk/README.md:54]: typo: below
